Run3Detector/analysis/utilities/milliqanProcessor.py

```diff
@@ -10,14 +12,96 @@
 
 class milliqanProcessor():
 
-    def __init__(self, filelist, branches, schedule=None, cuts=None, plotter=None, max_events=None):
+    def __init__(self, filelist, branches, schedule=None, cuts=None, plotter=None, max_events=None, runQualityOverride=False, qualityLevel="tight"):
+        self.qualityLevelString = qualityLevel
+        self.runQualityOverride = runQualityOverride
+        
+        #Converting the quality level to an integer
+        self.qualityDict = {"single_trigger": -1, "loose": 0, "medium": 1, "tight": 2}
+        if self.qualityLevelString not in self.qualityDict.keys():
+            raise Exception("Quality level '{0}' not recognized".format(self.qualityLevelString))
+        print("Chosen quality level: ", self.qualityLevelString)
+        self.qualityLevel = self.qualityDict[self.qualityLevelString]
+        
+        #Checks the filelist against goodRuns.json
         self.filelist = filelist
+        self.fileChecker() 
+        
         self.branches = branches
         self.mqSchedule = schedule
         #self.mqCuts = cuts
         #self.plotter = plotter
         self.max_events = max_events
 
+    #Get rid of the strings and make a dictionary so that it's easier to debug
+    def fileChecker(self):
+        #goodJson_array = ak.from_json(pathlib.Path("../goodRunTools/goodRunsMerged.json"))
```

Comment:
just add an if statement that if qualityLevel = "override" to return out of the function therefore keeping the run list intact